### PR TITLE
✨ Improved license check

### LINKED ISSUE #2159: Improve Score Reporting: License check regexes should be improved
License check should include the following file names: GPL-2.0, LICENSE, LICENCE, LICENSE.txt, LICENSE.rst, LICENSE.PSF, LICENSE.APACHE, LICENSE.BSD, LICENSE.md, LICENSE-MIT.

### PR DESCRIPTION
**Pull request by spencerschrock**
#### What kind of change does this PR introduce?
Enhance license check to address #2159 

- [X] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
GPL-2.0 not detected as a valid license. The other license files in #2159 are already accepted

#### What is the new behavior (if this is a feature change)?**
GPL-2.0 license now detected.

- [X] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
Fixes #2159 

#### Special notes for your reviewer

Should GPL-3.0 be included as well? Should be an easy regex change

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
GPL-2.0 license files are now detected by the License check
```

### CODE REVIEW

**Comment by spencerschrock** [comment]
Just to draw attention to the special instruction section:

> Should GPL-3.0 be included as well? Should be an easy regex change

**Review by naveensrinivasan — APPROVED** [review]
Cool! Thanks @spencerschrock

**Comment by naveensrinivasan** [comment]
> Just to draw attention to the special instruction section:
> 
> > Should GPL-3.0 be included as well? Should be an easy regex change

I would say Yes. But I am not a license expert. @david-a-wheeler would you be able to answer the question?

**Comment by spencerschrock** [comment]
I was also thinking yes, as the point of the License check is to see if a project has **published** a license. Not that it's an acceptable license. That's more for the ingesting project to determine if a license works for them. 

I'll go ahead and push a commit, but we can wait for David to chime in.

**Comment by azeemshaikh38** [comment]
Does this fully fix #2159 ? Seems like we would still miss `LICENCE `?

**Review by azeemshaikh38 — COMMENTED** [review]
Does this fully fix https://github.com/ossf/scorecard/issues/2159 ? Seems like we would still miss LICENCE ?

File: checks/raw/license.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ const (
 	anyExt       = ".[^./]"
 	ofl          = "ofl"
 	patents      = "patents"
+	gpl          = "gpl(v|-)\\d"
```
Comment (azeemshaikh38) [inline checks/raw/license.go:41]: I'm confused - the unit tests seem to suggest we already handle the cases specified in #2159 (i.e apart from GPL no other change was added here). Then why did our cron job not catch these files?

**Comment by spencerschrock** [comment]
> Does this fully fix #2159 ? Seems like we would still miss LICENCE ?

I added a test for it here: https://github.com/ossf/scorecard/pull/2179/files#diff-7f01c46823a0fd163cfcb213880f051df6931d5ddb495f16beececaae085c17eR45.

It was covered by existing regex though:
```
	license      = "(un)?licen[sc]e"
	...
	regexChecks = []checks{
	...
	{rstr: license, f: nil},
```

File: checks/raw/license.go (hunk 1 above)
Comment (spencerschrock) [reply checks/raw/license.go:41]: Yes, most things were already captured by the existing behavior, at least at the level of `checkLicense`.

Let me confirm with the cron results

Comment (azeemshaikh38) [reply checks/raw/license.go:41]: I misread the paper, thanks for pointing this out @spencerschrock

**Review by azeemshaikh38 — APPROVED** [review]

**Pull request merged**